src/ol/interaction/select.js

```diff
@@ -97,7 +97,8 @@ ol.interaction.Select = function(opt_options) {
     style: options.style ? options.style :
       ol.interaction.Select.getDefaultStyleFunction(),
     updateWhileAnimating: true,
-    updateWhileInteracting: true
+    updateWhileInteracting: true,
+    zIndex: goog.isDef(options.zIndex) ? options.zIndex : Infinity
```

Comment:
Please use `options.zIndex !== undefined` instead of `goog.isDef(options.zIndex)`.